bugbug/models/performancebug.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+
+import xgboost
+from imblearn.over_sampling import BorderlineSMOTE
+from imblearn.pipeline import Pipeline as ImblearnPipeline
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.pipeline import Pipeline
+
+from bugbug import bug_features, bugzilla, feature_cleanup, utils
+from bugbug.model import BugModel
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def _is_performance_bug(bug_data) -> bool:
+    keywords = bug_data.get("keywords", [])
+    whiteboard = bug_data.get("whiteboard", "").lower()
+    cf_performance = bug_data.get("cf_performance", "")
+
+    performance_keywords = ("perf", "topperf", "main-thread-io")
+
+    whiteboard_performance_text = (
+        "fxperf",
+        "fxperfsize",
+        "snappy",
+        "pdfjs-c-performance",
+        "pdfjs-performance",
+        "sp3",
+    )
+
+    if any(
+        keyword.startswith(performance_keyword)
+        for keyword in keywords
+        for performance_keyword in performance_keywords
+    ):
+        return True
+
+    if any(f"[{text}" in whiteboard for text in whiteboard_performance_text):
+        return True
+
+    if cf_performance and cf_performance not in ("---", "?"):
+        return True
```

Comment:
Updated:

https://github.com/mozilla/bugbug/blob/64384ebf6e02af1f8b80e4d1f25fb8e0cfed2c5a/bugbug/models/performancebug.py#L24-L25

Reply:
The logic in `bug_to_types` could be refactored to better fit our use case. These are the main points:

- It would be best to avoid creating models that depend on each other. If there is a need to modify the logic in the `bugtype` model, it might not be apparent that other models will be affected. Therefore, it is better to keep the common logic separate from the model file.
- We don't need to check all types in this case, just the performance type. So, we can split the function and only use the part that detects if the bug is related to performance.

I will file a separate issue for this, as it does not need to be done as part of this PR.

Reply:
I filed a follow-up issue: https://github.com/mozilla/bugbug/issues/3900